### PR TITLE
8289004: investigate if SharedRuntime::get_java_tid parameter should be a JavaThread*

### PR DESCRIPTION
**Pull request by robehn**
Hi, please consider.

Yes it should be JavaThread*.
But some additional places needed to use JavaThread*

Passes t1-3.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289004](https://bugs.openjdk.org/browse/JDK-8289004): investigate if SharedRuntime::get_java_tid parameter should be a JavaThread*

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10532/head:pull/10532` \
`$ git checkout pull/10532`

Update a local copy of the PR: \
`$ git checkout pull/10532` \
`$ git pull https://git.openjdk.org/jdk pull/10532/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10532`

View PR using the GUI difftool: \
`$ git pr show -t 10532`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10532.diff">https://git.openjdk.org/jdk/pull/10532.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/runtime/sharedRuntime.cpp [hunk 1]
```diff
@@ -996,15 +996,15 @@ JRT_ENTRY_NO_ASYNC(void, SharedRuntime::register_finalizer(JavaThread* current,
...
-              "current cannot touch oops after its GC barrier is detached.");
-    oop obj = JavaThread::cast(thread)->threadObj();
-    return (obj == NULL) ? 0 : java_lang_Thread::thread_id(obj);
+jlong SharedRuntime::get_java_tid(JavaThread* thread) {
+  assert(thread != NULL, "No thread");
+  if (thread == NULL) {
+    return 0;
   }
-  return 0;
+  guarantee(Thread::current() != thread || JavaThread::cast(thread)->is_oop_safe(),
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/sharedRuntime.cpp:1004]: You don't need the cast any more.

File: src/hotspot/share/runtime/sharedRuntime.cpp [hunk 2]
```diff
@@ -996,15 +996,15 @@ JRT_ENTRY_NO_ASYNC(void, SharedRuntime::register_finalizer(JavaThread* current,
...
+            "current cannot touch oops after its GC barrier is detached.");
+  oop obj = JavaThread::cast(thread)->threadObj();
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/sharedRuntime.cpp:1006]: Again no cast needed.

**Review by dholmes-ora — APPROVED** [review]
Changes look fine, but some additional changes needed in `get_java_tid`.

Thanks.

File: src/hotspot/share/runtime/sharedRuntime.cpp (hunk 1 above)
Comment (robehn) [reply src/hotspot/share/runtime/sharedRuntime.cpp:1004]: Thanks, fixing!

File: src/hotspot/share/runtime/sharedRuntime.cpp (hunk 2 above)
Comment (robehn) [reply src/hotspot/share/runtime/sharedRuntime.cpp:1006]: Thanks, fixing!

**Review by dholmes-ora — APPROVED** [review]

**Comment by robehn** [comment]
Thanks David!

**Review by jdksjolen — APPROVED** [review]
This looks good to me, thanks for the fix.

**Comment by robehn** [comment]
Thanks!

**Pull request closed without merging**